include/libcyphal/transport/udp/tx_rx_sockets.hpp

```diff
@@ -0,0 +1,109 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_UDP_TX_RX_SOCKETS_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_UDP_TX_RX_SOCKETS_HPP_INCLUDED
+
+#include "libcyphal/transport/errors.hpp"
+#include "libcyphal/transport/types.hpp"
+#include "libcyphal/types.hpp"
+
+#include <cetl/pf17/cetlpf.hpp>
+#include <udpard.h>
+
+#include <cstddef>
+#include <cstdint>
+
+namespace libcyphal
+{
+namespace transport
+{
+namespace udp
+{
+
+/// @brief Defines IP endpoint information in use for multicast transmissions.
+///
+struct IpEndpoint final
+{
+    std::uint32_t ip_address;
+    std::uint16_t udp_port;
+};
```

Comment:
I think it would be wiser to alias `UdpardUDPIPEndpoint` so that you don't have to convert there and back in your code. Also I think it's more accurate to call it a UDP/IP endpoint, not just an IP endpoint.

Reply:
My thinking was that we are under `udp::` namespace, so the type is kinda already has UDP, like `upd::IpEndpoint`...